src/tracks/InstrumentTrack.cpp

```diff
@@ -126,6 +128,25 @@ InstrumentTrack::InstrumentTrack( TrackContainer* tc ) :
 	}
 
 
+	// Initialize the m_midiCCEnabled variable, but it's actually going to be connected
+	// to a LedButton
+	m_midiCCEnable = new BoolModel( false, NULL, "Enable/Disable MIDI CC" );
+
+	// Initialize the MIDI CC controller models and connect them to the method that processes
+	// the midi cc events
+	QSignalMapper *midiCCSignalMapper = new QSignalMapper( this );
+
+	for( int i = 0; i < MIDI_CC_MAX_CONTROLLERS; ++i )
+	{
+		m_midiCCModel[i] = new FloatModel(0.0f, 0.0f, 127.0f, 1.0f, NULL, QString("CC Controller ") + QString::number(i));
+		connect( m_midiCCModel[i], SIGNAL( dataChanged() ),
+			midiCCSignalMapper, SLOT( map() ) );
```

Comment:
You can use a lambda function here like this:
 https://github.com/LMMS/lmms/blob/619b1d67cd6f6361d2546ce7b8e157e7598d702d/plugins/VstEffect/VstEffectControls.cpp#L93-L94

Reply:
I think you need to add `Qt::DirectConnection`. Without this, the UI thread will handle automated CC changes, which is not desired.
There's a problem, `processOutEvent` is realtime unsafe. However, LMMS is already calling it on realtime threads, so I think it's not a big problem here.

Reply:
My first attempt was using lambda functions, but I was running into compilation errors and just now reading your comment I realized why. I was still using the `SIGNAL()` macro, so `connect` expected a `const char *` where I used a lambda function (probably has to do with the `SIGNAL/SLOT` shenanigans). I changed the second parameter to `&FloatModel::dataChanged` and it compiled fine. I'll soon update the PR with the change!

Btw, I got a little curious, `processOutEvent` currently opens space for some race-conditions? I'd guess it being called so often would make us see these type of bugs at least some times. So in the future we might have to implement a `mutex` on it?

Reply:
Just updated the code to use lambda functions instead, I'll ping you later on Discord to ask more about the thread safeness of `processOutEvent`!